nbextensions/usability/hide_input_all/main.js

```diff
@@ -10,8 +10,13 @@ define([
     "use strict";
 
     function set_input_visible(show) {
-        IPython.notebook.metadata.hide_input = show;
```

Comment:
I think simply ` IPython.notebook.metadata.hide_input = !show;`should be enough.